### PR TITLE
No longer shows success message if content moving is cancelled

### PR DESCRIPTION
**Pull request by miguelcrpinto**
### Prerequisites

- [x ] I have added steps to test this contribution in the description below

If there's an existing issue for this PR then this fixes [issue 13923](https://github.com/umbraco/Umbraco-CMS/issues/13923)

### Description
- Added AttemptMove method to the ContentService
- Updated ContentController PostMove method to return ValidationProblem whenever the node is not moved

### CODE REVIEW

**Comment by emmagarland** [comment]
Hi @miguelcrpinto!

Thanks for your PR to fix #13923, where a success message shows even if content moving is cancelled. 

One of the Core Collaborators team will review this as soon as possible 👍 

Best wishes

Emma

File: src/Umbraco.Core/Services/ContentService.cs [hunk 1]
```diff
@@ -2442,20 +2442,25 @@ public OperationResult MoveToRecycleBin(IContent content, int userId = Constants
...
-            return;
+            return OperationResult.Attempt.Succeed(eventMessages);
         }
 
         // if moving to the recycle bin then use the proper method
         if (parentId == Constants.System.RecycleBinContent)
         {
             MoveToRecycleBin(content, userId);
-            return;
+            return OperationResult.Attempt.Succeed(eventMessages);
```
Comment (lauraneto) [inline src/Umbraco.Core/Services/ContentService.cs:2461]: Shouldn't we use the return from the `MoveToRecycleBin`?

**Review by lauraneto — CHANGES_REQUESTED** [review]
Hi @miguelcrpinto ,

Sorry for the delay!
Unfortunately it's a bit too late to get this to v13, so we have two options:
1. Change the return type of the `Move` method and target this to v14.
2. Make this non breaking. At the moment your changes are breaking because you have added a new method to an existing public interface.

If you decide to make it non breaking you would have to:
- Have an additional method with a non conventional name (as you already did), but having a default implementation on the interface
- (Optionally) Mark the old method as obsolete (as it doesn't really make sense to keep both methods) and change all of its references to use the new one

Let me know what your thoughts are on this one 😉
Also tagging @nul800sebastiaan for his thoughts.

**Comment by miguelcrpinto** [comment]
@lauraneto isn't V14 using different API methods in the backoffice? I can take a look at V14 and see what's being done there and also check if this is still an issue there.

This PR was mean to be a stepping stone to fix issue [13922](https://github.com/umbraco/Umbraco-CMS/issues/13922) but for this issue I'd have to change the API methods return types and the backoffice angular code. The API methods change would also be breaking changes. If 14 uses new API methods, I wonder if it is still worth to fix these.

**Comment by lauraneto** [comment]
Hi @miguelcrpinto ,

Sorry for the delay!
I tested the initial issue in V14 and good news is that it is fixed there 🙌, by the introduction a breaking change like we had discussed above.
As we cannot apply the same fix for <V13, and V13 is LTS, I think we can go for the (temporary) non breaking route.
I made some slight changes to the code, let me know what you think!

**Comment by Matthew-Wise** [comment]
@miguelcrpinto thank you for your PR, sorry its taken a while to get this merged

**Pull request merged**